### DISCUSSION TITLE
DPMS monitor sleep toggle

### DISCUSSION THREAD

**Discussion by cathoderaydude** [86Box/86Box#4841] [Ideas]
Searched for discussions on this subject, didn't find anything, might not have used the right terms. Maybe there's even a setting for this and I didn't see it, but: is it possible for 86box to ignore OS requests to put the display to sleep?

This absolutely falls in the "doctor it hurts when I do this" "then don't do that" category, but I have problems remembering to do stuff, and when I'm reinstalling OSes over and over while experimenting, I forget to disable display sleep in power management every time. The end result is that a lot of Windows 9x machines I set up (including some real ones) have issues with shutting the screen off when the machine is left idle, then going into a coma.

I leave for a few hours, come back, and the machine won't respond. It's going on with a VM I have open right now (cfg attached, in case you're curious.) Was the hang actually _caused_ by DPMS? Well, maybe it's a display driver bug, or maybe something else entirely, but it'd be nice to see the last frame of the desktop instead of a black screen just in case there's some hint of what went wrong.

I'm probably just picking bad drivers, and if I simply turn power management off it won't be an issue. But it occurs to me that I don't want _any VM_ to do this, ever, regardless of use case or OS. Display sleep just doesn't serve a purpose in a VM, so crash or no crash, it'd be great if I could disable it completely - it would help with things like video captures of long running processes where I'm not necessarily paying attention to the VM constantly. I'd put the setting in a template config so it replicates to every new machine I set up, and that would paper over my memory issues.

But is this even possible? Can a graphics card refuse to put the monitor to sleep, or does it have to happen at the driver level? Would it require edits to every single video device implementation separately? Is this remotely worth the effort? Guessing the answer is "no", but I figured it was worth asking.

Also guessing that at least some OSes turn the screen black in software, just in case DPMS doesn't work, so the whole thing is probably moot. I don't know how to test for that though; maybe someone else knows if there's even a point to this. Thanks for reading!

[86box.txt](https://github.com/user-attachments/files/17082969/86box.txt)

**Comment by OBattler** [comment]
This is a known bug, I believe @richardg867 is investigating it already (he's the one who wrote the DPMS code).
